src/main/scala/pt/tecnico/dsi/keystone/services/CRUDService.scala

```diff
@@ -62,11 +62,17 @@ abstract class CRUDService[F[_], T: Encoder: Decoder]
     case response => F.raiseError(UnexpectedStatus(response.status))
   }
 
-  /*{
-    "error" : {
-      "message" : "Conflict occurred attempting to store user - Duplicate entry found with name teste at domain ID default.",
-      "code" : 409,
-      "title" : "Conflict"
-    }
-  }*/
+  def disable(id: String)(implicit ev: T <:< WithEnabled[T]): F[Unit] =
+    updateEnable(id, false)
+
+  def enable(id: String)(implicit ev: T <:< WithEnabled[T]): F[Unit] =
+    updateEnable(id, true)
+
+  private def updateEnable(id: String, value: Boolean)(implicit ev: T <:< WithEnabled[T]) : F[Unit] = {
+    for {
+      obj <- get(id)
+      _ <- update(obj.id, ev(obj.model).withEnabled(value))
+    } yield ()
```

Comment:
```suggestion
    get(id).flatMap { obj => 
      update(obj.id, ev(obj.model).withEnabled(value))
    }
```

Reply:
This is simpler

Reply:
Não compila porque o método retorna `F[Unit]`